### ISSUE TITLE
Rig stuck in DNS resolve problem...

### ISSUE THREAD

**Issue by papampi** [papampi/nvOC_by_fullzero#94]
My internet went down for 5 minutes and after it came back some of my rigs got stuck in a name resolution problem.
Miner log:

```
Cannot resolve 'europe.ethash-hub.miningpoolhub.com'
ETH: Stratum - Cannot connect to europe.ethash-hub.miningpoolhub.com:20535
ETH: Stratum - Failed to connect, retry in 5 sec...
Cannot resolve 'europe.ethash-hub.miningpoolhub.com'
ETH: Stratum - Cannot connect to europe.ethash-hub.miningpoolhub.com:20535
ETH: Stratum - Failed to connect, retry in 5 sec...
Cannot resolve 'europe.ethash-hub.miningpoolhub.com'
ETH: Stratum - Cannot connect to europe.ethash-hub.miningpoolhub.com:20535
ETH: Stratum - Failed to connect, retry in 5 sec...

```
Watchdog log:
```
WARNING: Mon Jun 11 22:26:08 +0430 2018 - Internet is down, checking again in 30 seconds...
nc: getaddrinfo: Temporary failure in name resolution
nc: getaddrinfo: Temporary failure in name resolution
WARNING: Mon Jun 11 22:27:18 +0430 2018 - Internet is down, checking again in 30 seconds...
nc: getaddrinfo: Temporary failure in name resolution
nc: getaddrinfo: Temporary failure in name resolution
WARNING: Mon Jun 11 22:28:28 +0430 2018 - Internet is down, checking again in 30 seconds...
nc: getaddrinfo: Temporary failure in name resolution
nc: getaddrinfo: Temporary failure in name resolution
WARNING: Mon Jun 11 22:29:38 +0430 2018 - Internet is down, checking again in 30 seconds...

```

I knew internet was up because I was connected to the rigs via ssh from home, then I restarted miner on one of them and it solved the problem. this happened on 5 rigs and all with same symptom.
So I thought we should change how watchdog check network and react.
May be 1st ping check google dns, then check google.com with "nc" and if ping was up but nc couldn't resolve restart miner or ... 

Any better idea?

**Comment by papampi** [comment]
So It was weird,
Network had more problems last night and only those same 5 rigs was showing that symptom.
Added "-minspeed 50" to claymore opts so it restart miner if hash was low
No idea what is the problem.

**Comment by LuKePicci** [comment]
No ideas about that, The error may be thrown if you setup a DNS server which is remote (eg. google's 8.8.8.8).  Does restarting a single miner on one rig solved the problem of all 5 rigs or you needed to restart all of them?

**Comment by papampi** [comment]
All the rigs are almost identical and with static IP.
Just checked /etc/resolvconf/resolv.conf.d/base and only 1 rig has nameservers set in it, all others including those 5 are empty.

At first occurrence I restarted miners with killing watchdog and 3main, but when it happened again, I just killed miner on one rig and waited, after 10-15 seconds watchdog was able to resolve google address and goes for internet is up, checked miner not running and started it.
Waited for other rigs a while, but had to kill all other miners and their watchdogs start resolving address and starting miners.

**Comment by doftorul** [comment]
I can attest to this being the case for several hundred of rigs, and its a documented ubuntu bug. I will come back with the bug bulletin shortly.

**Comment by papampi** [comment]
I have 15 rigs, and only 5 showing this problem.

**Comment by LuKePicci** [comment]
The workaround is now available from #116
